crates/curp/src/server/curp_node.rs

```diff
@@ -753,23 +749,9 @@ impl<C: Command, CE: CommandExecutor<C>, RC: RoleChange> CurpNode<C, CE, RC> {
         curp: Arc<RawCurp<C, RC>>,
         cmd_executor: Arc<CE>,
         as_rx: flume::Receiver<TaskType<C>>,
-        shutdown_listener: Listener,
     ) {
-        #[allow(
-            clippy::arithmetic_side_effects,
-            clippy::ignored_unit_patterns,
-            clippy::pattern_type_mismatch
-        )]
-        // introduced by tokio select
-        loop {
-            tokio::select! {
-                _ = shutdown_listener.wait() => {
-                    break;
-                }
-                Ok(task) = as_rx.recv_async() => {
-                    Self::handle_as_task(&curp, &cmd_executor, task).await;
-                }
-            }
+        while let Ok(task) = as_rx.recv_async().await {
```

Comment:
This change seems related to some previous changes?
Could we ensure that when cluster shutdown, all senders will be dropped?

Reply:
This task could be directly canceled without waiting for the senders to be dropped. Because the tasks in the channel are already stored in the curp logs and will be re-executed during next recovery.

Reply:
But since you remove the `shutdown_listener`, this while-let loop will jump out only when all senders are dropped. ([flume::Receiver::recv_async](https://docs.rs/flume/latest/flume/struct.Receiver.html#method.recv_async), if my comprehension is not wrong)

Reply:
During the node shutdown, it will be canceled after it yield to the runtime.